lib/publiccloud/azure.pm

```diff
@@ -61,65 +70,110 @@ sub get_image_id {
     return $self->SUPER::get_image_id($img_url);
 }
 
-=head2 generate_azure_image_definition
+=head2 img_blob_exists
 
-    my $definition = $self->generate_azure_image_definition();
+Finds the image blob in Azure
 
-Generated the Azure Image name from the job settings. If present, it takes the PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION setting.
-If not present it generated the image definition name based on distri, version, flavor and SKU.
+ * Requires image name parameter, e.g. 'SLES15-SP5-BYOS.aarch64-1.0.0-Azure-Build1.68.xz'
+ * Returns 0 if the image blob  has not been found or 1 if the image blob has ben found.
 
-Note: Image definitions needs to be distinct names and can only serve one architecture!
-
-Example: 'SLE-MICRO-5.4-BYOS-AZURE-X86_64-GEN2'
 =cut
 
-sub generate_azure_image_definition {
-    my $arch = (get_var('PUBLIC_CLOUD_ARCH', get_required_var('ARCH'))) eq 'x86_64' ? 'x64' : 'Arm64';
-    my $definition = get_required_var('DISTRI') . '-' .
-      get_required_var('VERSION') . '-' .
-      get_required_var('FLAVOR') . '-' .
-      $arch . '-' .
-      get_var('PUBLIC_CLOUD_AZURE_SKU', $default_sku);
-    return get_var("PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION", uc($definition));
-}
-
-sub find_img {
+sub img_blob_exists {
     my ($self, $name) = @_;
-    my ($json, $md5, $image);
-
-    return if (!$self->resource_exist());
-
-    $name = $self->get_image_name($name);
+    my $md5;
 
     my $storage_account = get_var('PUBLIC_CLOUD_STORAGE_ACCOUNT', 'eisleqaopenqa');
     my $key = $self->get_storage_account_keys($storage_account);
+
     my $container = $self->container;
 
-    $json = script_output("az storage blob show --account-key $key -o json " .
+    # TODO: This is for transition period only. We used to upload blobs such as:
+    # SLES15-SP5-SAP.x86_64-1.0.0-Azure-Build1.78-V1.vhd but the '-V1' at the end is not needed
```

Comment:
I am not sure about this part . At least I will double check this with Robert before changing this part . **technically** you totally right image for gen1 and gen2 binary identical . BUT again **technically** SUSE still uploading TWO separate images for gen1 and gen2 . So : 
1. Tests always try as much as possible mimic behavior of SUT so if SUT is uploading two images we should do the same no matter what 
2. By removing this division here we taking responsibility of keep track on the fact that gen1 and gen2 images still binary identical . Keeping in mind that Public Cloud developers already shipping two separate images this will be really hard to catch and I am afraid no one will warn us about such change if/when it will happen